tests/test_controller.py

```diff
@@ -780,6 +781,19 @@ def test_authenticated_patron_root_lane(self):
                 eq_(302, response.status_code)
                 eq_("http://cdn/default/groups/", response.headers['location'])
 
+    def test_authentication_document(self):
+        """Test the ability to retrieve an Authentication For OPDS document."""
+        with self.request_context_with_library(
+                "/", headers=dict(Authorization=self.invalid_auth)):
```

Comment:
Note that we get the authentication document even though we provided invalid authentication credentials. The authentication document itself does not require credentials to access.